components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -0,0 +1,2222 @@
+         * Chgrp action passes only when light admin has Chgrp permission
+         * and is not Sudoing (Sudoing can be thought of as destroyer of the
+         * privileges, because it forces the permissions of the Sudoed-as user
+         * onto the light admin).*/
+        boolean chgrpNoSudoExpectSuccessAnyGroup = !isSudoing && permChgrp;
+        /* When data owner is member of the target group,
+         * Chgrp action passes also when light admin is
+         * Sudoed as the data owner */
+        boolean isExpectSuccessInMemberGroup = chgrpNoSudoExpectSuccessAnyGroup || isSudoing;
+        /* create a Dataset as the normalUser and import into it */
+        loginUser(normalUser);
+        Dataset dat = mmFactory.simpleDataset();
+        Dataset sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
+        final RString imageName = omero.rtypes.rstring(fakeImageFile.getName());
+        final List<List<RType>> result = iQuery.projection(
+                "SELECT id FROM OriginalFile WHERE name = :name ORDER BY id DESC LIMIT 1",
+                new ParametersI().add("name", imageName));
+        final long previousId = result.isEmpty() ? -1 : ((RLong) result.get(0).get(0)).getValue();
+        List<String> path = Collections.singletonList(fakeImageFile.getPath());
+        importFileset(path, path.size(), sentDat);
+        final List<RType> resultAfterImport = iQuery.projection(
+                "SELECT id, details.group.id FROM OriginalFile o WHERE o.id > :id AND o.name = :name",
+                new ParametersI().addId(previousId).add("name", imageName)).get(0);
+        final long remoteFileId = ((RLong) resultAfterImport.get(0)).getValue();
+        long remoteFileGroupId = ((RLong) resultAfterImport.get(1)).getValue();
+        assertOwnedBy((new OriginalFileI(remoteFileId, false)), normalUser);
+        Assert.assertEquals(remoteFileGroupId, normalUser.groupId);
+        Image image = (Image) iQuery.findByQuery(
+                "FROM Image WHERE fileset IN "
+                + "(SELECT fileset FROM FilesetEntry WHERE originalFile.id = :id)",
+                new ParametersI().addId(remoteFileId));
+        /* set up the light admin's permissions for this test */
+        List<String> permissions = new ArrayList<String>();
+        permissions.add(AdminPrivilegeSudo.value);
+        if (permChgrp) permissions.add(AdminPrivilegeChgrp.value);
+        final EventContext lightAdmin;
+        lightAdmin = loginNewAdmin(true, permissions);
+        try {
+            sudo(new ExperimenterI(normalUser.userId, false));
+            }catch (SecurityViolation sv) {
+            }
+        /* take care of workflows which do not use sudo */
+        if (!isSudoing) {
+            loginUser(lightAdmin);
+        }
+        /*in order to find the image in whatever group, get context with group
+         * set to -1 (=all groups) */
+        client.getImplicitContext().put("omero.group", Long.toString(-1));
```

Comment:
fixed in https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/2cbddc73fc35666b53d3d626813a38fac1ec688f